### ISSUE TITLE
Add configuration option to disable plugin dynamically

### ISSUE THREAD

**Issue by autonomousapps** [KeepSafe/dexcount-gradle-plugin#254]
In one of my projects, I have the following:
```groovy
if (isOnCi()) {
    // This plugin adds ~5s to every build
    apply plugin: 'com.getkeepsafe.dexcount'
}
```
I am currently converting my build scripts to use the `plugins {}` block, but that block does not support custom logic like this. It would be nice if I could move this plugin to that block, but then configure it in the `dexcount {}` block based on arbitrary logic.

Actually, I just figured out how to accomplish this with the following:
```groovy
tasks.withType(com.getkeepsafe.dexcount.DexMethodCountTaskBase) {
    // This plugin adds ~5s to every build
    onlyIf { isOnCi() }
}
```
but it seems a bit fragile by comparison to first-class support in the `dexcount {}` block.

**Comment by autonomousapps** [comment]
You're probably going to find more people using it now that Gradle's Kotlin
DSL is nearing 1.0 status. Using the plugins block gives a Kotlin script
stronger type support than the old apply method. Just an FYI.

**Comment by benjamin-bader** [comment]
Yes, that makes sense - it will be good to be ready for the 1.0 release.

**Comment by benjamin-bader** [comment]
I've just released version 0.8.4 to Maven Central, which includes the new configuration option.  It should be available within the next 15 minutes.
